src/form_components/repeating_group.coffee

```diff
@@ -101,7 +101,7 @@ FormRenderer.Views.ResponseFieldRepeatingGroup = Backbone.View.extend
     _.last(@views).focus()
 
   removeEntry: (e) ->
-    idx = @$el.find('.js-remove-entry').index(e.target)
+    idx = @$el.find('.js-remove-entry').index(e.target.closest('.js-remove-entry'))
```

Comment:
The clicked target here is actually the `i` tag wtihin the `js-remove-entry` wrapper, so we're traversing back up to the `js-remove-entry` element for comparison instead of comparing it against an icon.